Add sort keys option for dataclasses inheriting from mixin

mishamsk opened the pull request:
Hi,

Thought this is small enough to directly open a PR and see if you would support this feature.

What: add a `sort_keys` option for `BaseConfig` which would efficiently generate serialization code such that keys are sorted on creation (which in many cases will make much heavier encoder approach unnecessary).

What's missing: 
- I've only added the feature to Mixin subclasses for now, but not for arbitrary dataclasses
- I haven't added any docs, since there is no guarantee you would approve this PR
- I've only added one test, maybe you would want additional ones

Rationale:
I am using key sort (via `__post_serialize__` [hook](https://github.com/mishamsk/pyoak/blob/9c69c5dd0f4bc55be0db6d02e866d121904a333e/src/pyoak/serialize.py#L95)) a lot in order to maintain stable serialized view of my objects (those pesky trees). This would make it more efficient (depending on how classes are defined of course).

Fatal1ty commented on mashumaro/core/meta/code/builder.py:
```diff
@@ -818,7 +818,11 @@ def _add_pack_method_lines(self, method_name: str) -> None:
             aliases = {}
             fields_could_be_none = set()
-            for fname, ftype in field_types.items():
+            fnames_and_types = list(field_types.items())
```
We don't need a list if `sort_keys` is not set:
```python
fnames_and_types = field_types.items()
if self.get_config().sort_keys:
    fnames_and_ftypes = sorted(
        fnames_and_ftypes, key=lambda x: x[0]
    )  # type: ignore
```

mishamsk replied:
@Fatal1ty thanks. not a big fan of untyped code, so I used a typed version of your suggestion

Fatal1ty reviewed (changes requested):
There is one minor but everything else is fine.

> I've only added the feature to Mixin subclasses for now, but not for arbitrary dataclasses

It will also work for arbitrary dataclasses:

```python
@dataclass
class Foo:
    b: int
    a: int

@dataclass
class Bar(DataClassDictMixin):
    z: int
    foo: Foo

bar = Bar(z=3, foo=Foo(b=2, a=1))
print(bar.to_dict())  # {'foo': {'a': 1, 'b': 2}, 'z': 3}
```

Would you like to add such a case to the tests since you mentioned it?

mishamsk commented:
> Would you like to add such a case to the tests since you mentioned it?
forgot to pull latest changes, so naturally it didn't work for dataclasses in my branch.

now pulled and rebased. added a test + added a README

mishamsk commented:
glad to help!

Fatal1ty commented:
@mishamsk 

I just realised that I was wrong. Config options are not propagated to inner dataclasses by design, and the tests didn't show it because two dictionaries with different key order are equal. I created the pull request to fix the tests.

mishamsk commented:
@Fatal1ty ouch. sorry. my bad!
